http-core/src/main/scala/org/apache/pekko/http/impl/model/parser/HeaderParser.scala

```diff
@@ -86,8 +86,9 @@ private[http] class HeaderParser(
   def failure(error: Throwable): HeaderParser.Failure =
     HeaderParser.Failure {
       error match {
-        case IllegalUriException(info) => info
-        case NonFatal(e)               => ErrorInfo.fromCompoundString(e.getMessage)
+        case IllegalUriException(info)           => info
+        case NonFatal(e) if e.getMessage == null => ErrorInfo.fromCompoundString(e.toString)
```

Comment:
Can use eq

Reply:
`e.getMessage eq null` ? I don't think it is  good. `eq` is used to replace the equals method, and when compared to null, it is usually to use `==`